Sources/NIO/FileHandle.swift

```diff
@@ -81,12 +81,50 @@ public final class NIOFileHandle: FileDescriptor {
 }
 
 extension NIOFileHandle {
+    /// `Mode` represents file access modes.
+    public enum Mode {
```

Comment:
just a note, this could be an `OptionSet`?

Reply:
It could, but I don't think it makes much sense, what would combining `.readOnly` and `.readWrite` mean for example? And combining `.readOnly` and '.writeOnly` gives you `.readWrite`, if I'm not mistaken.

Reply:
well but you could just have `.read` and `.write`. Then you would have

- `[.read, .write]` --> `O_RDWR`
- `[.read]` --> `O_RDONLY`
- `[.write]` --> `O_WRONLY`

Reply:
What could work. Converted to `OptionSet`